### ISSUE TITLE
Duplicate keys detected

### ISSUE THREAD

**Issue by gjportegies** [SnowdogApps/magento2-menu#228]
Magento: 2.4.3
Snowdog Menu: 2.18.0

This is quite an edge case but I ran into the following issue: 

<img width="941" alt="Schermafbeelding 2022-01-07 om 11 16 05" src="https://user-images.githubusercontent.com/6707410/148530884-f0c445be-a3dc-45f0-9775-bee91a1a24e9.png">

My console log is full with these messages and the autocomplete dropdown gets stuck. The shop we're running is a migrated from Magento 1 and has the following setup:
- Website 1
  - Nederlands
- Website 2
  - Dutch
  - English
- Website 3
  - Dutch

The stores Nederlands, Dutch and Dutch have a lot of CMS pages with exactly the same name and I this is probably what is causing the errors. In my case this is only happening with CMS page's but I think this can also be reproduced with blocks and categories.

I managed to workaround the issue by adding the page ID in the `Model/NodeType/CmsPage.php` file.

```diff
    /**
     * @inheritDoc
     */
    public function fetchConfigData()
    {
        $this->profiler->start(__METHOD__);

        $options = array_map(function ($page) {
            return [
+               'id' => $page->getId(),
                'label' => $page->getTitle(),
                'value' => $page->getIdentifier(),
                'store' => array_filter(
                    (array)$page->getStoreId(),
                    function ($id) {
                        return (int)$id !== 0;
                    }
                )
            ];
        }, $this->getResource()->fetchConfigData());

        $this->profiler->stop(__METHOD__);

        return $options;
    }
```

**Comment by gjportegies** [comment]
Hi @andrzejborkowski I’ve already submitted a PR https://github.com/SnowdogApps/magento2-menu/pull/229

**Comment by jochhop** [comment]
This issue has been resolved in PR #229 , so I am closing it , thanks @gjportegies for opening and resolving the issue with your contribution
